src/parser/core/Combatant.ts

```diff
@@ -46,7 +46,7 @@ class Combatant extends Entity {
     return this._combatantInfo.player;
   }
 
-  get spec(): Spec | undefined {
```

Comment:
this change is not correct. there are combatants with missing spec info, mostly cases with missing `combatantinfo` events. it happens much more often on Classic than Retail

Reply:
@emallson I did a Helper type and enforced that "Selected Combat" is always that type, and we can access `spec` without having to check null. 

I did the assertion all the way in `Combatant.ts` so we could assert it once and not have to recheck in `get`-ers

I'm sure there's more properties we could assert in this helper type.

Have a look at the commit please! 64c387422ad92ae8bb44ed81a05b91146260878e

Reply:
Ooof this breaks all the integration tests because eventhough [`TestCombatLogParser` overrides selectedCombatant](https://github.com/oBusk/WoWAnalyzer/blob/31fb77a36c62cdd0713818fd8b7d53019e05cdb7/src/parser/core/tests/TestCombatLogParser.ts#L33), the `Combatants` module runs in whole, meaning it will fail... Not sure what would be the correct approach.. Any thoughts?

Reply:
Looks like the [Combatants](https://github.com/oBusk/WoWAnalyzer/blob/primary-stat/src/parser/shared/modules/Combatants.ts#L57) module does not use `selectedCombatant` to get the selected combatant

Reply:
No exactly, normally it's vice versa `selectedCombatant()` uses `Combatants.selected`. In the dummy combatlogparser we try to ignore what the `Combatants` module is doing by just overriding `selectedCombatant`, but `Combatants` will still run.

Reply:
yea in this case it looks like it the `this.players` array is not initialized in a way that functions with the `TestCombatLogParser`.

for the sake of making it function with current tests, you could make it just skip the check if the combatant isn't defined (`this.players[...] === undefined`). Not ideal but retrofitting a mock combatant onto every test would also be unpleasant.

Reply:
I guess more "correct" would either be to mock the combatantinfo or mock the combatant module itself? But I'm not deep enough into the module system to know how to do that. 

It seems to work with this little check if the combatant is defined, do you feel this is correct? Or should we avoid messing with Combatant module and just doing some type assertion in CombatLogparser instead.

Reply:
Since this case comes up for the tests only, I think its ok if we cheat a little. Our test infra as a whole needs love following a bunch of the work from this expansion but hasn't gotten it yet

Reply:
The e2e tests that was failing was because of the changes to the Combatants module, so I figured the `SelectedCombatant` change was more hassle than it was worth and rolled it back, and added a helper getter which assures that `primaryStat` is available (or die trying)